strfry/plugins/policies.ts

```diff
@@ -7,19 +7,45 @@ import {
   rateLimitPolicy,
   readStdin,
   writeStdout,
-} from "https://gitlab.com/soapbox-pub/strfry-policies/-/raw/develop/mod.ts";
+} from "https://raw.githubusercontent.com/planetary-social/strfry-policies/refs/heads/ban-interval/mod.ts";
 import nosPolicy from "./nos_policy.ts";
 
 const localhost = "127.0.0.1";
 const eventsIp = await getEventsIp();
+const one_minute = 60 * 1000;
+const two_days = 2 * 24 * 60 * one_minute;
 
 // Policies that reject faster should be at the top. So synchronous policies should be at the top.
 const policies = [
   nosPolicy,
   [hellthreadPolicy, { limit: 100 }],
+
   // Async policies
   [antiDuplicationPolicy, { ttl: 60000, minLength: 50 }],
-  [rateLimitPolicy, { whitelist: [localhost, eventsIp] }],
+
+  // For abusers, we ban the ip after 20 requests per minute, ban ip for 2 days
+  // Remember to leave the more loose rate limit policy at the top
+  [
+    rateLimitPolicy,
+    {
+      max: 20,
+      interval: one_minute,
+      ban_interval: two_days,
+      whitelist: [localhost, eventsIp],
+      // We use a different db url so that this limiter is not affected by the other limiters
+      databaseUrl: "sqlite:///tmp/banning-strfry-rate-limit-policy.sqlite3",
```

Comment:
Won't this be cleared whenever the docker container is updated? I think it should be in a directory that is persisted.

Reply:
Good point, I can use the db dir that is already a volume